plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotConnectorSmokeTest.java

```diff
@@ -1727,6 +1728,18 @@ SELECT COUNT(*)
                 WHERE double_col NOT IN (0.0, -16.33, -17.33)"""))
                 .matches("VALUES (BIGINT '7')")
                 .isNotFullyPushedDown(FilterNode.class);
+
+        assertThat(query("SELECT price, vendor FROM " + JSON_TABLE + " WHERE price IS NOT NULL"))
+                .containsAll("VALUES (REAL '3.5', VARCHAR 'vendor1')")
+                .isFullyPushedDown();
+
+        assertThat(query("SELECT COUNT(*) FROM " + JSON_TABLE + " WHERE price IS NOT NULL"))
+                .matches("VALUES (BIGINT '7')")
+                .isFullyPushedDown();
+
+        assertThat(query("SELECT COUNT(*) FROM " + JSON_TABLE + " WHERE price IS NULL"))
+                .matches("VALUES (BIGINT '0')")
+                .isFullyPushedDown();
```

Comment:
Maybe do this for a copy of the table w null handling enabled? Or you can choose a simpler table (`hybrid` is pretty simple). 
Also: what is the behavior of a hybrid table where null handling is enabled for realtime but disabled for offline or vice versa?